File: dynamic programming/0-1knapsack.java
import java.util.*;
public class ClassicalKnapsack
{
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int n = scanner.nextInt();
		int[] weight = new int[n];
		int[] value = new int[n];
		System.out.println("Enter the weight:-");
		for(int i = 0; i < n; i++){
		    weight[i] = scanner.nextInt();
		}
		System.out.println("Enter the value:-");
		for(int i = 0; i < n; i++){
		    value[i] = scanner.nextInt();
		}
		System.out.println("Enter the required capacity:-");
		int w = scanner.nextInt();
		int[][] dp = new int[n+1][w+1];
		for(int i = 0; i <= n; i++){
		    for(int j = 0; j <= w; j++){
		        if(i == 0 || j == 0){
		            dp[i][j] = 0;
		        }
		        else if(weight[i-1] <= j){
		            dp[i][j] = Math.max(value[i-1]+dp[i-1][j-weight[i-1]], dp[i-1][j]);
		        }
		        else{
		            dp[i][j] = dp[i-1][j];
		        }
		    }
		}
		//print the output
		System.out.println(dp[n][w]);
	}
}

File: dynamic programming/c++/0-1Knapsack.cpp
#include <bits/stdc++.h>
using namespace std;
int knapSack(int W, int wt[], int val[], int n)
{
	// making and initializing dp array
	int dp[W + 1];
	memset(dp, 0, sizeof(dp));

	for (int i = 1; i < n + 1; i++) {
		for (int w = W; w >= 0; w--) {

			if (wt[i - 1] <= w)
				// finding the maximum value
				dp[w] = max(dp[w],
							dp[w - wt[i - 1]] + val[i - 1]);
		}
	}
	return dp[W]; // returning the maximum value of knapsack
}
int main()
{   
    int n ; 
    cout<<"Enter number of items" ; 
    cin>>n ; 
    int va[n] ,wt[n]; 
    cout<<"Enter Value of items\n" ;
    for(int i = 0 ; i<n ;i++)
    {
        cin>>val[i] ; 
    }
    cout<<"Enter Weight of items\n" ;
    for(int i = 0 ; i<n ;i++)
    {
        cin>>wt[i] ; 
    }
    cout<<"Enter capacity of Knapsack\n" ; 
    int W ; 
    cin>>W ; 
	cout << knapSack(W, wt, val, n);
	return 0;
}
